rest/src/db/dbUtils.js

```diff
@@ -77,6 +80,31 @@ const dbUtils = {
 			return { [sortFieldDbRelation[options.sortField]]: { [1 === options.sortDirection ? '$gt' : '$lt']: offset } };
 		}
 		return undefined;
+	},
+
+	/**
+	 * Convert binary to resolved address
+	 * @param {MongoDb.Binary} binary Address|NamespaceId from MongoDb.
+	 * @returns {string} AddressBase32|NamespaceId
+	 */
+	bufferToUnresolvedAddress: binary => {
+		if (!(binary instanceof MongoDb.Binary))
+			return undefined;
+
+		const hex = binary.toString('hex');
+		const bit0 = convert.hexToUint8(hex.substr(1, 2))[0];
+
+		if (16 === (bit0 & 16)) {
+			// only 8 bytes are relevant to resolve the NamespaceId
+			const namespaceId = hex.substr(2, 16);
+
+			// retun as namespace Id
+			return convert.uint8ToHex(convert.hexToUint8Reverse(namespaceId));
+		}
+
+		// return as Address base 32
+		const hexToUint8 = convert.hexToUint8(hex);
+		return address.addressToString(hexToUint8);
```

Comment:
why this instead of always addressToString? that is how we normally represent aliases, i think?

Reply:
cc @gimer

Reply:
for my understanding, `addressToString` method is only way convert Uint8 to an encoded address.

Reply:
i meant returning alias as namespace id seems strange to me. i don't think we use that representation anywhere else, so i'd _just_ use `addressToString`

Reply:
I got what you mean now, the reason to add namespace id  is to handle `Alias address`.

This is the simple. 

![image](https://user-images.githubusercontent.com/5649156/128860066-9d22e8eb-3fd3-41f8-a614-ee27b7076a74.png)

http://ngl-dual-501.testnet.symboldev.network:3000/transactions/confirmed/3D81C7E22AE0504ABD86EBBBF998E8A1F1F9E6D4C329523ED31760D93B8735C0

Reply:
@gimer - what are your thoughts on this?

Reply:
ugh, I misread the code this is _not_ ok, this method returns two completely different objects :/

Reply:
What about if Rest returns either the encoded address or the namespace id hex (like in other places) when working on unresolved addresses?

```
"recipientAddress": "NATNE7Q5BITMUTRRN6IB4I7FLSDRDWZA35C4KNQ", <-real address
"recipientAddress": "291F837E059AE13C", <-Alias
```

Related PR:

https://github.com/symbol/symbol-openapi/pull/272/files#diff-299dfd070f15158c760575d45de8aeb702e4820df3540fbecbe34bcc15f88695R3

Reply:
What is the expected result, if that is namespace Id hex on `recipientAddress`?

or we should not allow namespace id appear on "recipientAddress"?

Reply:
i like rg911 suggestion to return object with encoded address and optional alias namespace id. or even just always returning encoded address might be sufficient.

i think it would be confusing for consumers to return either encoded address or namespace id.

Reply:
I believe our goal is to solve the confusion problem for the API.
I will support always returning encoded addresses for the address field will be great.

We need another PR for this.
 - Always resolve the `alias` to `address`